File: UF1_Proyecto_compose/app/src/main/java/com/example/uf1_proyecto_compose/data/task/TaskDAO.java
package com.example.uf1_proyecto_compose.data.task;

import java.util.List;

public interface TaskDAO {

    List<Task> getLocalTasks();

    void storeLocalTasks(Task task);

    List<Task> getRemoteTasks();

    void storeRemoteTask(Task task);

}

File: UF1_Proyecto_compose/app/src/main/java/com/example/uf1_proyecto_compose/data/database/local/dao/TaskDao.kt
package com.example.uf1_proyecto_compose.data.database.local.dao

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.example.uf1_proyecto_compose.data.database.local.entity.TaskEntity

@Dao
interface TaskDao {

    @Query("SELECT * FROM tasks")
    suspend fun getAllTasks(): List<TaskEntity>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertTasks(tasks: List<TaskEntity>)

    @Query("DELETE FROM tasks")
    suspend fun deleteAll()

    @Delete
    suspend fun delete(task: TaskEntity)
}